### ISSUE TITLE
Use of an `@deprecated` property is not reported as an error

### ISSUE THREAD

**Issue by ottumm** [microsoft/pyright#6094]
The following typechecks cleanly with pyright 1.1.330 (cli), but it should report an error for use of a deprecated property.

```python
from typing_extensions import deprecated


class Foo:
    @property
    @deprecated("Don't use this")
    def bar(self) -> str:
        return "hello"


f = Foo()
f.bar
```

```toml
[tool.pyright]
pythonVersion = "3.12"
include = ["projects", "src", "script", "typings"]
strict = ["projects", "src", "script", "typings"]

reportImportCycles = true
reportUnnecessaryTypeIgnoreComment = true
```

I tried moving the `@deprecated` decorator above the `@property` decorator but that didn't make a difference.

**Comment by erictraut** [comment]
Thanks for the bug report. I've added support for `@deprecated` when applied to property accessors and descriptor functions (`__get__`, `__set__`, and `__delete__`). This will be included in the next release.

**Comment by ottumm** [comment]
That’s excellent thank you so much!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.331, which I just published. It will also be included in a future release of pylance.
